src/xsyspath.cpp

```diff
@@ -10,32 +10,51 @@
 
 #include "pybind11/pybind11.h"
 
-#include <iostream>
-#include <regex>
 #include <string>
+#include <fstream>
 
 #include "xeus-python/xeus_python_config.hpp"
 
 #include "xsyspath.hpp"
 #include "xpaths.hpp"
 
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <array>
+
+#include "pybind11/eval.h"
+
+namespace py = pybind11;
+
 namespace xpyt
 {
+    std::string exec(const std::string& command)
+    {
+        std::array<char, 128> buffer;
+        std::string result;
+        std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(command.c_str(), "r"), pclose);
+        if (!pipe)
+        {
+            throw std::runtime_error("popen() failed!");
+        }
+        while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr)
+        {
+            result += buffer.data();
+        }
+        return result;
+    }
+
     void set_syspath()
     {
-// The XEUS_PYTHON_SYSPATH compile-time definition can be used.
-// to specify the syspath locations as relative paths to the PREFIX.
-#if defined(XEUS_PYTHON_SYSPATH)
-        static const std::string syspath = std::regex_replace(
-            XPYT_STRINGIFY(XEUS_PYTHON_SYSPATH),
-            std::regex("\\{PREFIX\\}"),
-            prefix_path());
-#if PY_MAJOR_VERSION == 2
-        Py_SetPath(const_cast<char*>(syspath.c_str()));
-#else
-        static const std::wstring wstr(syspath.cbegin(), syspath.cend());;
-        Py_SetPath(const_cast<wchar_t*>(wstr.c_str()));
-#endif
-#endif
+        std::ifstream host_file(prefix_path() + "/share/xeus-python/host");
+        std::string host_python((std::istreambuf_iterator<char>(host_file) ),
+                                (std::istreambuf_iterator<char>()));
+        host_python.pop_back(); // remove newline
```

Comment:
Maybe we could do something more elegant...

I am hesitant to make the `host` file a JSON file, which would prevent this issue, and give room to more keys.